### PR TITLE
Call updater purge on package removal

### PR DESCRIPTION
**Pull request by raphaelgavache**
### What does this PR do?

Add a purge command to delete all files created by the updater, especially systemd units. Purge never errors and prints warn when the commands fails to execute.

Following, the updater purge needs to be called on prerm called with a `remove` argument[package removal doc](https://debian-handbook.info/browse/stable/sect.package-meta-information.html)
```
5.2.2.2. Package Removal
The steps to remove a package are analogous to the installation steps. The main difference is that the removal scripts of the package are called:
dpkg calls the prerm script and passes the remove argument.
dpkg removes all of the package's files, with the exception of the configuration files and maintainer scripts.
dpkg executes the postrm script and passes remove as argument. Afterwards, all of the maintainer scripts, except the postrm script, are removed. If the user has not used the “purge” option, the process stops here.
```

### CODE REVIEW

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -81,6 +82,18 @@ func Bootstrap(ctx context.Context, pkg string) error {
 	return u.Bootstrap(ctx, pkg)
 }
 
+// Purge removes files installed by the updater
+func Purge() {
+	// golanci on windows somehow expects a return value
+	service.RemoveAgentUnits() // nolint:errcheck
```
Comment (BaptisteFoy) [inline pkg/updater/updater.go:88]: On windows [RemoveAgentUnits](https://github.com/DataDog/datadog-agent/blob/5a2a21c211c1b735fab7c3294f1247eeb9169368/pkg/updater/service/datadog_agent_windows.go#L27) has an error as a return value, which is why the CI triggered. Let's remove it and have the same signature

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/updater/updater.go [hunk 2]
```diff
@@ -81,6 +82,18 @@ func Bootstrap(ctx context.Context, pkg string) error {
...
+	if os.RemoveAll(defaultLocksPath) != nil {
```
Comment (arbll) [inline pkg/updater/updater.go:89]: Let's log the returned error

File: pkg/updater/updater.go (hunk 1 above)
Comment (arbll) [reply pkg/updater/updater.go:88]: Linux has an error as the return value as well. Let's just log the error, we want to be able to debug this in e2e and with customers

**Review by arbll — APPROVED** [review]
lgtm after comments are addressed

File: pkg/updater/updater.go (hunk 1 above)
Comment (raphaelgavache) [reply pkg/updater/updater.go:88]: there's no error on linux, thanks for catching the windows
On purpose I'm passing through each command to cleanup everything that is possible, and log warns for each error seen
https://github.com/DataDog/datadog-agent/pull/23597/files#diff-105824131aeb51aea4c79cc098a114e7346fbbaeb2d709e24ac878eedc596e4eR80-R112

**Pull request merged**